### PR TITLE
fix(linter): revert unmatched rule error

### PR DESCRIPTION
**Pull request by camchenry**
- related to https://github.com/oxc-project/oxc/issues/6988

This is causing a lot of errors currently such as https://github.com/oxc-project/oxc/issues/7233. I think we need to handle rule names more robustly first:

- https://github.com/oxc-project/oxc/issues/7240
- https://github.com/oxc-project/oxc/issues/7082
- https://github.com/oxc-project/oxc/discussions/7242

then, I think we can revisit this and maybe implement it as an actual lint plugin too?  https://github.com/oxc-project/oxc/discussions/7086

### CODE REVIEW

**Comment by camchenry** [comment]
* **#7257** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7257?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: oxlint.config.json
Comment (camc314) [inline oxlint.config.json:1]: revert?

**Review by camc314 — APPROVED** [review]

File: oxlint.config.json
Comment (camchenry) [reply oxlint.config.json:1]: lol, oops. I was using this for testing.

**Comment by camchenry** [comment]
Noting that we are still accumulating the unknown rules internally, but we just won't report it to users anymore.

**Pull request merged**